src/Analyzers/Core/Analyzers/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs

```diff
@@ -111,17 +120,9 @@ private void AnalyzeSemanticModel(SemanticModelAnalysisContext context)
         {
             var tree = context.SemanticModel.SyntaxTree;
             var cancellationToken = context.CancellationToken;
+            var language = context.SemanticModel.Language;
 
-            if (!(context.Options is WorkspaceAnalyzerOptions workspaceOptions))
-            {
-                return;
-            }
-
-            var language = context.SemanticModel.Compilation.Language;
-            var service = workspaceOptions.Services.GetLanguageServices(language)
-                                                   .GetService<IUnnecessaryImportsService>();
-
-            var unnecessaryImports = service.GetUnnecessaryImports(context.SemanticModel, cancellationToken);
+            var unnecessaryImports = UnnecessaryImportsProvider.GetUnnecessaryImports(context.SemanticModel, cancellationToken);
```

Comment:
can we not make this change in another PR?  why does this need to move down with everything else?  This seems to have a larger scope of change that makes it harder to validate with everything else.